### ISSUE TITLE
Adjoining lists do not parse correctly

### ISSUE THREAD

**Issue by MGChecker** [Python-Markdown/markdown#1058]
Example:
```
Some text

* List
* List

* List
* List
* List

Some further text
```
If you write this, I would expect it to be parsed to two unordered lists. In practice, the first element of the second list is parsed to its own list though.

**Comment by mitya57** [comment]
Hi!

This is described in https://python-markdown.github.io/#differences. Quoting from there:

> __Consecutive Lists__
>
> While the syntax rules are not clear on this, many implementations (including the original) do not end one list and start a second list when the list marker (asterisks, pluses, hyphens, and numbers) changes. For consistency, Python-Markdown maintains the same behavior with no plans to change in the foreseeable future. That said, the [Sane List Extension](https://python-markdown.github.io/extensions/sane_lists/) is available to provide a less surprising behavior.

**Comment by MGChecker** [comment]
This does not seem to be related to the issue I am facing, that lists are split up with no apparent reason.

I forgot to mention that I am actually using the `sane_lists` extension though.

**Comment by mitya57** [comment]
So do you mean that you get **three** lists?

I think what you are actually getting is one list, just the second and third items are wrapped in a paragraph (`<p>`):

For me, the output with Python-Markdown 3.3.3 and `sane_lists` extension enabled is as follows:
```html
<p>Some text</p>
<ul>
<li>List</li>
<li>
<p>List</p>
</li>
<li>
<p>List</p>
</li>
<li>List</li>
<li>List</li>
</ul>
<p>Some further text</p>
```

If you are getting different HTML, please paste it here and specify the version you are using.

**Comment by waylan** [comment]
@MGChecker I assume you are concluding that you are getting multiple separate lists because you see some gaps of whitespace between the items in your browser. To properly understand what is going on, we need you to provide the HTML. I expect you will find that the HTML is exactly as @mitya57 provided. You appear to not be familiar with the difference between loose lists and tight lists. I suggest reading the [Syntax Rules][1], particularly the second half of the section on lists. Also, note that the sane lists extension does not alter the behavior of loose/tight lists.

As a summary:

## Tight lists

A "tight list item" is a list item which is not wrapped in a paragraph (`<p>`) in the HTML output. A tight list is created by not including blank lines between items. 

For example this list:

    * item one
    * item two

... is rendered as this HTML:

```html
<ul>
  <li>item one</li>
  <li>item two</li>
</ul>
```

Note that no `<p>` tags appear in that list. It renders in the browser as:

> * item one
> * item two

Notice how close together the list items are. Using my browser's "inspect" tool, I can also see that the margin and padding of the `<li>` element are both set to `0`.

## Loose lists

A "loose list item" is a list in which each item is wrapped in a paragraph (`<p>`) in the HTML output. A loose list is created by including a blank line between each list item.

For example this list:

    * item one

    * item two

... is rendered as this HTML:

```html
<ul>
  <li>
    <p>item one<p>
  </li>
  <li>
    <p>item two<p>
  </li>
</ul>
````

Note that the contents of each list item `<li>` is wrapped in a `<p>` tag. This generally causes additional whitespace to display in the browser.  It renders in the browser as:

> * item one
> 
> * item two

Notice how close together the list items are. Using my browser's "inspect" tool, I can also see that the `<p>` element has both a margin-top and `margin-bottom` of `16px`, which adds extra whitespace between the items. Of course, that specific CSS is defined here on GitHub's site. Other sites may have different values. However, GitHub has used the typical browser default in this case.

## Mixed Lists

A "mixed list" is a list which has both "loose list items" and "tight list items". Note that only the items adjacent to a blank line at loose list items.  All other items are tight list items.. Consider this example:

```
* tight item 1
* tight item 2
* loose item 1

* loose item 2
* tight item 3
* tight item 4
```

That renders as this HTML

``` html
<ul>
  <li>tight item 1</li>
  <li>tight item 2</li>
  <li>
    <p>loose item 1</p>
  </li>
  <li>
    <p>loose item 2</p>
  </li>
  <li>tight item 3</li>
  <li>tight item 4</li>
</ul>
```

Note that `loose item 1` and `loose item 2`, both of which were adjacent to the blank line, are wrapped in `<p>` tags, while the other elements are not. That is displayed in the browser as:

<blockquote>
<ul>
<li>tight item 1</li>
<li>tight item 2</li>
<li>
<p>loose item 1</p>
</li>
<li>
<p>loose item 2</p>
</li>
<li>tight item 3</li>
<li>tight item 4</li>
</ul>
</blockquote>

Finally, I will note that this behavior exactly matches the reference implementation, as you can confirm [here][2].

Note that the sane lists extension does not alter this behavior of lists. It only alters the behavior of combining different list types (ordered and unordered) into one type.

## Why do we have both tight and loose lists?

It has been my observation that most people only need tight lists most of the time. However, loose lists are needed when you want to nest block level items within a list. As a reminder, you can nest code blocks and blockquotes within a list item. Therefore, within that same list item, plain blocks of text would need to be wrapped in paragraphs. And of course, paragraphs need a blank line between them. Consider this example:

```
* list item

* list item with nested content.

    * nested list item 1
    * nested list item 2

        a nested code block

    > a nested blockquote

    a nested paragraph

* list item
```

While it was not necessary to include blank lines between the middle item and the first and third items, it is better to do so as that will result in every item in the list have a consistent appearance. No one ever really wants a "mixed list' so I recommend to never use them unless you really know what you are doing.

## But how do I make adjacent lists?

In short, you don't. Anything more than a single blank line is completely ignored by Markdown. And as a single blank line doesn't end a list, you must always include something which is not a list between to separate lists.

While I understand that some of these behaviors may not be intuitive, we didn't make the rules or the reference implementation. We simply follow them. I suppose an extension which alters the behavior might be desired by some, but I don't have a need for it myself.

I'm closing this as 'wont fix'.

[1]: https://daringfireball.net/projects/markdown/syntax#list
[2]: https://daringfireball.net/projects/markdown/dingus
